app/mailers/user_mailer.rb

```diff
@@ -8,68 +8,21 @@ class UserMailer < ActionMailer::Base
     default from: AppConfig.first.admin_email, cc: AppConfig.first.admin_email
   end
 
-  def checkin_receipt(reservation)
+  # checks the status of the current reservation and sends the appropriate email
+  def reservation_status_update(reservation)
     set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Your equipment return receipt')
-  end
-
-  def checkout_receipt(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Your equipment checkout receipt')
-  end
-
-  def missed_reservation_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation Missed')
-  end
-
-  def overdue_checkin_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] OVERDUE: equipment checkin')
-  end
 
-  def overdue_checked_in_fine(overdue_checked_in)
-    return if overdue_checked_in.equipment_model.late_fee == 0
-    set_app_config
-    @overdue_checked_in = overdue_checked_in
-    mail(to: overdue_checked_in.reserver.email,
-         subject: '[Reservations] Overdue equipment fine')
-  end
+    return if reservation.status == 'returned overdue' &&
+              reservation.equipment_model.late_fee == 0
 
-  def reservation_confirmation(complete_reservation)
-    set_app_config
-    @complete_reservation = complete_reservation
-    mail(to: complete_reservation.first.reserver.email,
-         subject: '[Reservations] Confirmation of your reservation')
-  end
-
-  def upcoming_checkin_notification(reservation)
-    set_app_config
     @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reminder: equipment check in')
-  end
 
-  def request_approved_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation request approved!')
-  end
+    status_formatted = @reservation.status.split.map(&:capitalize) * ' '
 
-  def request_denied_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation request denied')
+    mail(to: @reservation.reserver.email,
+         subject: '[Reservations] ' \
+                  "#{@reservation.equipment_model.name.capitalize} "\
+                  " Status Update: #{status_formatted}")
```

Comment:
Hmmm, I'm not in love with this subject, it gets a little unwieldy. What about "[Reservations] `equipment model name` `status`". For example, "[Reservations] Nikon D90 Available Today", "[Reservations] iPad 2 Due Today", "[Reservations] Composite RCA Adapter Overdue". I changed "Starts Today" to "Available Today", but otherwise I think it works. What do you think?